src/api/base.ts

```diff
@@ -0,0 +1,39 @@
+const baseUrl = `http://localhost:8000/api/`;
+
+const apiUrl = (endpoint: string) => {
+    return `${baseUrl}${endpoint}`;
+};
+
+export const fetchToken = async (data: { username: string, password: string }) => {
+    const response = await fetch(apiUrl(`token/`), {
+        method: `POST`,
+        headers: {
+            "Content-Type": `application/json`
+        },
+        body: JSON.stringify(data)
+    });
+    const token = await response.json();
+
+    if (token.access) {
+        localStorage.setItem(`token`, token.access);
+    }
```

Comment:
I don't love this setup, but I think localStorage saving of token stuff should be in the api area, because only the api area needs info.